extension-push/src/push_utils.cpp

```diff
@@ -111,9 +122,6 @@ void dmPush::HandleCommand(dmPush::Command* cmd, void* ctx)
     }
     free((void*)cmd->m_Result);
     free((void*)cmd->m_Error);
-
-    if (cmd->m_Command == dmPush::COMMAND_TYPE_REGISTRATION_RESULT)
```

Comment:
Why here and not inside the command handler?!

Reply:
Several reasons I think:
* Keeping destruction code in one place
* Is it always the case that if the callback is "invalid", there should be no destruction on the callback?
* "HandlePushMessageResult" doesn't really seem like it would destroy the callback.
* "HandleCommand" deals with things for that struct, and the callback belongs to that struct

None of them are dealbreakers for me, however I'm more concerned about possibly the DestroyCallback might not always be called. Are we leaving references behind? I don't currently think so. What if the implementation of IsCallbackValid() and DestroyCallback() changes in the future? Worst case is that we will have to patch all extensions again I guess?

Reply:
Reading the code here:
https://github.com/defold/extension-push/blob/master/extension-push/src/push_android.cpp#L66

If the script is released between the "Register" and the callback, we'll call the handler, it will probably pass the IsValidCallback, but fail the SetupCallback. Then we'll skip the DestroyCallback, that will introduce a leak.

Reply:
Good points made there! I'll put the code back where it belongs :-)